### ISSUE TITLE
DurationField does not handle OverflowError if an out of range input is passed 

### ISSUE THREAD

**Issue by v1k45** [encode/django-rest-framework#5310]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](http://www.django-rest-framework.org/topics/third-party-resources/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce

1. Create a view and serializer which take duration as input using `DurationField`.
Test view and serializer used for reproducing this issue:

```python
from rest_framework import serializers, views

class DurationSerializer(serializers.Serializer):
    duration = serializers.DurationField()


class DRFDurationAPI(views.APIView):
    def post(self, request, *args, **kwargs):
        serializer = DurationSerializer(data=request.data)
        if serializer.is_valid(raise_exception=True):
            return Response({"success": True})

```

2. Send an _out of range_ value as input for duration field.
Example data

```json
{
	"duration": "1000000000 10:11:12"
}
```

## Expected behavior

Receive a ValidationError by view, similar to the validation error we get if the input format is incorrect:

```json
{"duration":["Duration is too large."]}
```

## Actual behavior

We get a 500 Error, with the following exception message:

```
OverflowError at /api/duration
days=1000000000; must have magnitude <= 999999999

Request Method: POST
Request URL: http://localhost:8000/api/duration
Django Version: 1.10.7
Python Executable: /home/vikas/.virtualenvs/venv/bin/python
Python Version: 3.5.3
Python Path: ['/home/vikas/code/work/venv', '/home/vikas/code/work/project', '/home/vikas/.virtualenvs/venv/lib/python35.zip', '/home/vikas/.virtualenvs/venv/lib/python3.5', '/home/vikas/.virtualenvs/venv/lib/python3.5/plat-x86_64-linux-gnu', '/home/vikas/.virtualenvs/venv/lib/python3.5/lib-dynload', '/usr/lib/python3.5', '/usr/lib/python3.5/plat-x86_64-linux-gnu', '/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages']
Server time: Sun, 6 Aug 2017 13:41:52 +0000
Installed Applications:
['django.contrib.sites',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'rest_framework',
]
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']


Traceback:  

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/django/core/handlers/exception.py" in inner
  42.             response = get_response(request)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/django/core/handlers/base.py" in _legacy_get_response
  249.             response = self._get_response(request)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/django/views/decorators/csrf.py" in wrapped_view
  58.         return view_func(*args, **kwargs)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/django/views/generic/base.py" in view
  68.             return self.dispatch(request, *args, **kwargs)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/rest_framework/views.py" in dispatch
  489.             response = self.handle_exception(exc)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/rest_framework/views.py" in handle_exception
  449.             self.raise_uncaught_exception(exc)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/rest_framework/views.py" in dispatch
  486.             response = handler(request, *args, **kwargs)

File "/home/vikas/code/work/venv/project/views.py" in post
  900.         if serializer.is_valid(raise_exception=True):

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/rest_framework/serializers.py" in is_valid
  237.                 self._validated_data = self.run_validation(self.initial_data)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/rest_framework/serializers.py" in run_validation
  432.         value = self.to_internal_value(data)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/rest_framework/serializers.py" in to_internal_value
  462.                 validated_value = field.run_validation(primitive_value)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/rest_framework/fields.py" in run_validation
  525.         value = self.to_internal_value(data)

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/rest_framework/fields.py" in to_internal_value
  1306.         parsed = parse_duration(six.text_type(value))

File "/home/vikas/.virtualenvs/venv/lib/python3.5/site-packages/django/utils/dateparse.py" in parse_duration
  127.         return datetime.timedelta(**kw)

Exception Type: OverflowError at /api/duration
Exception Value: days=1000000000; must have magnitude <= 999999999
```

If a more _long_ value is passed as days, the execption message changes to
```
Python int too large to convert to C long
```

A simple `try/except` statement in [`DurationField` ](https://github.com/encode/django-rest-framework/blob/master/rest_framework/fields.py#L1308) will fix this issue. I can send a pull request if this issue is accepted.

**Comment by rpkilby** [comment]
Hi @v1k45. This bug also exists with Django's `forms.DurationField`. I'm inclined to say that you should open a bug on Django's ticket tracker first, and then we'll follow suit.

**Comment by peterthomassen** [comment]
[Django fixed this in 2.1](https://github.com/django/django/commit/55b5393bd235aa6c55ea9a7da6f0e8b75b204084), but only for `forms.DurationField`.

It seems DRF's serializer doesn't go through that code path, so the problem is still there for DRF. I suggest reopening this.

Would a PR be welcome here that mirrors the fix for the Django form field?

**Comment by auvipy** [comment]
a PR to follow path of django compat is welcome
